### ISSUE TITLE
Dwire features

### ISSUE THREAD

**Issue by johcarvajal** [DelfiSpace/DWire#3]
Hello,
I would like to know what are the additional features that this library implements with respect to the Arduino Wire library?

**Comment by stefanlinden** [comment]
Hi Johan,

The idea is to have the main (user's) API as close to the original Wire library as possible. It should give the same results as when using pure Arduino. However, this library has been made specifically for the MSP432 microcontroller (for now), where Energia is used as the Arduino port.
Energia is as you know very powerful, but it has several issues in its Wire library on the MSP432 due to software driving the I2C communication instead of using the hardware modules (briefly mentioned in https://github.com/energia/Energia/issues/651). 
Firstly, the clock speed is limited to roughly 20 KHz. Secondly, the MSP432 is only able to work as an I2C master, able to interface with (3rd party) slave nodes on a bus. It is impossible to set the microcontroller up as a slave, even though the necessary function handles are in place. The interrupts will simply not trigger. Thirdly, it is not possible to add more than one I2C instance or changing which GPIO pins to use, possibly limiting the number of applications. Finally, during some simple testing we found several random instabilities in the Energia Wire library, possibly related to buffering problems.
The developers of Energia are currently working on these issues, but it is still unknown when and whether they will be fixed. Therefore, in the meantime, this library tries to fill the gap.
Unfortunately, to get the functionality, the DWire library must be used outside the Energia environment. However, using Code Composer Studio and the included example(s) as a baseline, the code complexity should still be able to be kept at a minimum.
In conclusion and to answer your question, this library does not add extra features compare to the Arduino Wire library per se, but it does add extra functionality otherwise expected from the Energia Wire port. The current HEAD (at time of writing) is able to configure an MSP432 as a master or slave, and transmit data from the former to the latter. Request functionality will be added very soon.

**Comment by johcarvajal** [comment]
Thanks for the clarification. One additional request. Is there a possibility to add the configuration setup process in the readme file, in order to start using your library with code composer?

**Comment by stefanlinden** [comment]
A short guide has been added in commit 76a4ffc.
